### DISCUSSION TITLE
Links in status messages not clickable but coincidentally contained Träwelling usernames clickable

### DISCUSSION THREAD

**Discussion by pascalwittler** [Traewelling/traewelling#3168] [Ideas / Feature Requests]
### Describe the bug

After posting a status message with a link to one of my own Mastodon posts [1], I noticed that the link was not clickable. My Mastodon username contained in the link is the same as my Träwelling username – and it turns out to be clickable with my Träwelling profile as the link target.

My expectation would be that text parts of status messages that can be parsed as valid URLs would be rendered as clickable links and they would be evaluated with higher priority than substrings.

[1] My status message including my clickable username and the unclickable link

![1](https://github.com/Traewelling/traewelling/assets/20669380/599722d1-7d1f-4f01-9b68-39a3902a2fb8)

### Steps to reproduce

1. Create a status message in your check-in
2. Insert a URL containing a valid Träwelling username

**Comment by MrKrisKrisu** [comment]
I'm not sure if I would really call that a “bug”. I wouldn't link external links.

However, the internal mention could perhaps be limited to the extent that it is only a mention if there are no other characters before or after.

**Comment by pascalwittler** [comment]
I definitely see your point.

Only evaluating a mention without non-space characters before and after it could lead in false negatives, for example in comma separated lists (as in `„On the way to event XY with @user1, @user2 and @user3“`).

Feel free to remove the „bug“ label. It was added automatically after I had to choose between creating an issue or a discussion by the forced templates.
